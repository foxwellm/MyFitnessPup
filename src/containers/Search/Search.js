import React, { Component } from 'react'
import { fetchDogs } from '../../thunks/fetchDogs'
import { connect } from 'react-redux'
import { Switch, Route, withRouter } from 'react-router-dom'
import { breeds } from '../../staticData/breeds'

export class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      zipCode: 77043,
      zipError: '',
      breeds: breeds,
      currentPage: 1,
      search: []
    }
  }


  handleChange = (e) => {

    this.setState({
      zipCode: e.target.value
    })
  }

  handleSearch = (e) => {
    e.preventDefault()
    const { search, breeds, zipCode } = this.state
    const { fetchDogs, storedDogs } = this.props

    if (search.length === 0) {

      breeds.forEach(dog => {
        if (!storedDogs[zipCode] || !storedDogs[zipCode][dog.breed]) {
          let options = {
            location: zipCode,
            // location: 77043,
            breed: dog.breed
          }
          fetchDogs(options)
        } else {
          console.log('there')
        }
      })
    }
  }

  render() {

    // const breeds = breeds
    // debugger

    return (
      <form onSubmit={this.handleSearch}>
        <input type='number' onChange={this.handleChange} placeholder='zip-code' name='zip' value={this.state.zipCode}></input>
        <button>Search</button>
        <div className='results-container'>
        
        </div>
      </form>
    )
  }
}

export const mapStateToProps = (state) => ({
  storedDogs: state.storedDogs
})

export const mapDispatchToProps = (dispatch) => ({
  fetchDogs: (options) => dispatch(fetchDogs(options))
})

export default connect(mapStateToProps, mapDispatchToProps)(Search)