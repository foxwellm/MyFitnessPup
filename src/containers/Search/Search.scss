.Search {
  height: calc(100% - 80px);
  display: flex;
  flex-direction: column;
}

.search-container {
  display: flex;
  overflow: scroll;
  height: 0px;
  transition: .15s;
  background-color: white;
}

.show {
  height: 200px;
  transition: .15s;
}

.trail-img {
  position: absolute;
  z-index: -1;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

.dog-loading {
  position: absolute;
  z-index: -1;
  top: 20%;
  left: 20%;
  height: 60%;
  width: 60%;
}

.search-bar {
  display: flex;
  justify-content: space-around;
  height: 30px;
  padding: 20px 0;
  position: relative;
  z-index: 2;
  background-color: white;
}

.results-container {
  flex: 1;
  position: relative;
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(360px, 1fr));
  grid-auto-rows: auto;
  // grid-gap: 1%;
  width: 90%;
  padding: 0 150px 0px 0;
  margin: auto;
  overflow-y: scroll;
}

.search-buttons-container {
  width: 100%;
  display: flex;
  justify-content: flex-end;
}

.next-prev-btn {
    text-decoration: none;
  cursor: pointer;
  transition: background-color 0.1s ease-out;
  text-align: center;
  border: 0;
  margin:20px;
  font-size: 15px;
  font-family: "Lato", sans-serif;
  color: #fcfcfc;
  background-color: salmon;
  height: 35px;
  width: 80px;
  border-radius: 3px;
  outline: none;
  &:hover {
    transform: scale(1.01);
    background-color: rgb(232, 99, 84);
    transition: background-color 0.1s ease-out;
  }
}

.search-input-container {
  display: flex;
  position: relative;
}

.search-num-display {
  font-size: 20px;
}

.search-error {
  position: absolute;
  top: 5px;
  left: -250px;
  color: red;
  font-size: 20px;
}

.search-input {
  font-size: 20px;
  height: 30px;
  padding-left: 8px;
  width: 92px;
}

.search-btn {
  text-decoration: none;
  cursor: pointer;
  transition: background-color 0.1s ease-out;
  text-align: center;
  border: 0;
  margin: 0 0 15px 10px;
  font-size: 15px;
  font-family: "Lato", sans-serif;
  color: #fcfcfc;
  background-color: #107568;
  height: 35px;
  width: 80px;
  border-radius: 3px;
  outline: none;
  &:hover {
    transform: scale(1.01);
    background-color: #005b4f;
    transition: background-color 0.1s ease-out;
  }
}

.specific-search-btn {
  width: 200px;
}


