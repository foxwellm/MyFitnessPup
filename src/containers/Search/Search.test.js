import React from 'react';
import { Search } from '../Search/Search'
import { shallow } from 'enzyme'
import { mapStateToProps, mapDispatchToProps } from '../Search/Search'

describe('Search', () => {
  let wrapper
  let mockFetchedDogs
  let mockStaticBreeds
  let mockStaticBreedInfo
  let shortID = require('short-id')

  beforeEach(() => {
    shortID.generate = jest.fn().mockImplementation(() => '63de45')
    mockFetchedDogs = {
      77043: {
        'Poodle': {
          cleanedDogs: [{ distance: '19 mi' }, { distance: '91 mi' }]
        },
        'Retriever': {
          cleanedDogs: [{ distance: '92 mi' }, { distance: '29 mi' }]
        }
      }
    }
    mockStaticBreeds = [
      { breed: 'Poodle', tag: 'poodle', img: 'poodle.jpg', isCold: true, isRunner: true, isClimber: true },
      { breed: 'Retriever', tag: 'retriever', img: 'retriever.jpg', isCold: true, isRunner: false, isClimber: true },
    ]
    mockStaticBreedInfo = [
      { name: 'Poodle', desc: "The Poodle was originally bred to pull sleds and carts long distances in the harsh Russian climate.They are powerfully built dogs with lively spirits, always ready for adventure at any time.A true endurance breed, Siberians make an excellent choice for those looking for a dog that will keep them entertained on long hikes.They are gentle and alert, but very social dogs that enjoy spending time outside." },
      { name: 'Retriever', desc: "The Retriever is an intelligent, adventurous breed that is always ready to go. Originally used as herding dogs, they still excel at this job today, as well as in many dog sports such as agility and obedience. Australian Shepherds make excellent hiking partners, as they are very athletic, friendly, and well suited for long hours of strenuous activity." },
    ]
    wrapper = shallow(<Search
      fetchedDogs={mockFetchedDogs}
      staticBreeds={mockStaticBreeds}
      staticBreedInfo={mockStaticBreedInfo}
    />)
  })
  it('should match the correct snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('handleChange', () => {
    const mockEvent = {
      target: {
        value: 77043
      }
    }
    it('should change state of zipCode', () => {
      wrapper.setState({ zipCode: 80204 })
      expect(wrapper.state('zipCode')).toEqual(80204)
      wrapper.instance().handleChange(mockEvent)
      expect(wrapper.state('zipCode')).toEqual(77043)
    })
  })

  // describe('updateCurrentSearchDogs', () => {

  //   it('should search all dogs when given none and sort by distance', () => {
  //     wrapper.setState({ zipCode: 77043 })
  //     wrapper.instance().updateCurrentSearchDogs()
  //     expect(wrapper.state('currentSearchDogs')).toEqual([{ distance: '19 mi' }, { distance: '29 mi' }, { distance: '91 mi' }, { distance: '92 mi' },])
  //   })

  //   it('should search dogs in search state and sort by distance', () => {
  //     wrapper.setState({ search: ['Retriever'], zipCode: 77043 })
  //     wrapper.instance().updateCurrentSearchDogs()
  //     expect(wrapper.state('currentSearchDogs')).toEqual([{ distance: '29 mi' }, { distance: '92 mi' },])
  //   })
  // })

  describe('handleSearch', () => {

    it('should set an error if zipcode is not valid', () => {
      const mockPreventDefault = {
        preventDefault: jest.fn()
      }

      wrapper.setState({ zipCode: 7704 })
      wrapper.instance().handleSearch(mockPreventDefault)
      expect(wrapper.state('zipError')).toEqual('Please enter valid zip code')
    })
  })

  describe('mapStateToProps', () => {

    it('should return an array of objects for storedDogs and boolean for isLoading and isDisplay', () => {
      const mockState = {
        staticBreeds: [{}, {}],
        fetchedDogs: [{}, {}],
        isLoading: true,
        isDisplay: true,
        ideas: [{}]

      }
      const expected = {
        staticBreeds: [{}, {}],
        fetchedDogs: [{}, {}],
        isLoading: true,
        isDisplay: true
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch when using retrieveDogs from MDTP', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.retrieveDogs(77043, ['Poodle', 'Retriever'])
      expect(mockDispatch).toHaveBeenCalled()
    })

    it('should call dispatch when using setLoading from MDTP', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setLoading(true)
      expect(mockDispatch).toHaveBeenCalled()
    })

    it('should call dispatch when using setDisplay from MDTP', () => {
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setDisplay(true)
      expect(mockDispatch).toHaveBeenCalled()
    })
  })

})
