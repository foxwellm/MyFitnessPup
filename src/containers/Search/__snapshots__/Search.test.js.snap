// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the correct snapshot 1`] = `
<div
  className="Search"
>
  <img
    alt="outdoor trail"
    className="trail-img"
    src="trail.jpg"
  />
  <div
    className="search-bar"
  >
    <div
      className="search-num-display"
    >
      Searching all dog breed(s)
      <button
        className="search-btn"
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <div
      className=""
    >
      <button
        className="search-btn specific-search-btn"
        onClick={[Function]}
      >
        Search specific breeds
      </button>
    </div>
    <form
      className="search-input-container"
      onSubmit={[Function]}
    >
      <div
        className="search-error"
      />
      <input
        className="search-input"
        name="zip"
        onChange={[Function]}
        placeholder="zip-code"
        type="number"
        value=""
      />
      <button
        className="search-btn"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="search-container "
  >
    <BreedCard
      active={false}
      breed="Poodle"
      cardNumber={0}
      handleSearchFilter={[Function]}
      img="poodle.jpg"
      isClimber={true}
      isCold={true}
      isRunner={true}
      key="6"
      tag="poodle"
    />
    <BreedCard
      active={false}
      breed="Retriever"
      cardNumber={1}
      handleSearchFilter={[Function]}
      img="retriever.jpg"
      isClimber={true}
      isCold={true}
      isRunner={false}
      key="6"
      tag="retriever"
    />
  </div>
</div>
`;

exports[`Search should match the correct snapshot when dogs are done loading from fetch 1`] = `
<div
  className="Search"
>
  <img
    alt="outdoor trail"
    className="trail-img"
    src="trail.jpg"
  />
  <div
    className="search-bar"
  >
    <div
      className="search-num-display"
    >
      Searching all dog breed(s)
      <button
        className="search-btn"
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <div
      className=""
    >
      <button
        className="search-btn specific-search-btn"
        onClick={[Function]}
      >
        Search specific breeds
      </button>
    </div>
    <form
      className="search-input-container"
      onSubmit={[Function]}
    >
      <div
        className="search-error"
      />
      <input
        className="search-input"
        name="zip"
        onChange={[Function]}
        placeholder="zip-code"
        type="number"
        value=""
      />
      <button
        className="search-btn"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="search-container "
  >
    <BreedCard
      active={false}
      breed="Poodle"
      cardNumber={0}
      handleSearchFilter={[Function]}
      img="poodle.jpg"
      isClimber={true}
      isCold={true}
      isRunner={true}
      key="6"
      tag="poodle"
    />
    <BreedCard
      active={false}
      breed="Retriever"
      cardNumber={1}
      handleSearchFilter={[Function]}
      img="retriever.jpg"
      isClimber={true}
      isCold={true}
      isRunner={false}
      key="6"
      tag="retriever"
    />
  </div>
  <div
    className="results-container"
    onScroll={[Function]}
  >
    <DogCard
      age="Young"
      breeds={Object {}}
      contact="petplace@gmail.com"
      description="Star is by far the star of the house.  We love her…ation please contact Alfred↵ alfredk9rr@gmail.com"
      distance="2.0 mi"
      gender="Female"
      id="37030888"
      key="6"
      link={
        Object {
          "href": "/v2/animals/436246",
        }
      }
      name="Star"
      photo="http://photos.petfinder.com/photos/pets/37030888/1/"
      size="Large"
      status="Available"
      url="https://www.petfinder.com/dog/vegas-357285"
      zip=""
    />
    <DogCard
      age="Baby"
      breeds={Object {}}
      contact="petplace@gmail.com"
      description="Butch is by far the star of the house.  We love her…ation please contact Alfred↵ alfredk9rr@gmail.com"
      distance="10.2 mi"
      gender="Male"
      id="37030889"
      key="6"
      link={
        Object {
          "href": "/v2/animals/436247",
        }
      }
      name="Butch"
      photo="http://photos.petfinder.com/photos/pets/37030888/1/"
      size="Large"
      status="Available"
      url="https://www.petfinder.com/dog/vegas-357285"
      zip=""
    />
    <div
      className="bot"
    >
      <div
        className="end-container"
      >
        No more dogs match your criteria
      </div>
    </div>
  </div>
</div>
`;

exports[`Search should match the correct snapshot when dogs are loading from fetch 1`] = `
<div
  className="Search"
>
  <img
    alt="outdoor trail"
    className="trail-img"
    src="trail.jpg"
  />
  <div
    className="search-bar"
  >
    <div
      className="search-num-display"
    >
      Searching all dog breed(s)
      <button
        className="search-btn"
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <div
      className=""
    >
      <button
        className="search-btn specific-search-btn"
        onClick={[Function]}
      >
        Search specific breeds
      </button>
    </div>
    <form
      className="search-input-container"
      onSubmit={[Function]}
    >
      <div
        className="search-error"
      />
      <input
        className="search-input"
        name="zip"
        onChange={[Function]}
        placeholder="zip-code"
        type="number"
        value=""
      />
      <button
        className="search-btn"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="search-container "
  >
    <BreedCard
      active={false}
      breed="Poodle"
      cardNumber={0}
      handleSearchFilter={[Function]}
      img="poodle.jpg"
      isClimber={true}
      isCold={true}
      isRunner={true}
      key="6"
      tag="poodle"
    />
    <BreedCard
      active={false}
      breed="Retriever"
      cardNumber={1}
      handleSearchFilter={[Function]}
      img="retriever.jpg"
      isClimber={true}
      isCold={true}
      isRunner={false}
      key="6"
      tag="retriever"
    />
  </div>
  <div
    className="results-container"
    onScroll={[Function]}
  >
    <DogCard
      age="Young"
      breeds={Object {}}
      contact="petplace@gmail.com"
      description="Star is by far the star of the house.  We love her…ation please contact Alfred↵ alfredk9rr@gmail.com"
      distance="2.0 mi"
      gender="Female"
      id="37030888"
      key="6"
      link={
        Object {
          "href": "/v2/animals/436246",
        }
      }
      name="Star"
      photo="http://photos.petfinder.com/photos/pets/37030888/1/"
      size="Large"
      status="Available"
      url="https://www.petfinder.com/dog/vegas-357285"
      zip=""
    />
    <DogCard
      age="Baby"
      breeds={Object {}}
      contact="petplace@gmail.com"
      description="Butch is by far the star of the house.  We love her…ation please contact Alfred↵ alfredk9rr@gmail.com"
      distance="10.2 mi"
      gender="Male"
      id="37030889"
      key="6"
      link={
        Object {
          "href": "/v2/animals/436247",
        }
      }
      name="Butch"
      photo="http://photos.petfinder.com/photos/pets/37030888/1/"
      size="Large"
      status="Available"
      url="https://www.petfinder.com/dog/vegas-357285"
      zip=""
    />
    <div
      className="bot"
    >
      <div
        className="end-container"
      >
        No more dogs match your criteria
      </div>
    </div>
  </div>
</div>
`;
